File: language/runtime/apple/core/Sources/RuntimeHostAppleCore/Input/Text/RuntimeHostTextInputEvent.swift
import Foundation

/// One Apple text input ingress event delivered into one runtime session.
public struct RuntimeHostTextInputEvent: Sendable, Hashable, Codable {
  /// The stable text session identifier.
  public let sessionID: UInt64
  /// The current text state reported by the Apple host.
  public let state: RuntimeHostTextInputState

  /// Create one text input event.
  public init(
    sessionID: UInt64,
    state: RuntimeHostTextInputState
  ) {
    self.sessionID = sessionID
    self.state = state
  }
}

File: language/runtime/android/kotlin/src/main/kotlin/dev/destack/runtime/android/input/text/RuntimeHostTextInputEvent.kt
package dev.destack.runtime.android.input.text

/**
 * One Android text input ingress event delivered into one runtime session.
 */
public data class RuntimeHostTextInputEvent(
    /**
     * The stable text session identifier.
     */
    val sessionId: Long,

    /**
     * The current text state reported by the Android host.
     */
    val state: RuntimeHostTextInputState,
)
